### PR TITLE
add qma6100p accelerometer

### PR DESCRIPTION
**Pull request by pborcin**
# Checklist for new Board Support package or Component

- [x] Component contains License
- [x] Component contains README.md
- [ ] Project [README.md](../README.md) updated
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to CI [upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml#L17)
- [x] New files were added to CI build job
- [ ] New BSP definitions added to [bsp_ext.py](../examples/bsp_ext.py)
- [ ] BSP was [added to SquareLine](https://github.com/espressif/esp-bsp/tree/master/SquareLine/common)
- [ ] New BSP supports IDF v4.4.x and later releases
- [x] _Optional:_ Component contains unit tests
- [x] CI passing

### CODE REVIEW

File: components/qma6100p/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+This driver, along with many other components from this repository, can be used as a package from [Espressif's IDF Component Registry](https://components.espressif.com). To include this driver in your project, run the following idf.py from the project's root directory:
+
+```
+    idf.py add-dependency "espressif/qma6100p"
+```
+
+Another option is to manually create a `idf_component.yml` file. You can find more about using .yml files for components from [Espressif's documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/tools/idf-component-manager.html).
+
+## See Also
+* [Sensors example](https://github.com/espressif/esp-bsp/tree/master/examples/display_sensors)
```
Comment (tore-espressif) [inline components/qma6100p/README.md:34]: Please remove this link. QMA6100 is not used in that example

File: components/qma6100p/test/qma6100p_test.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-FileCopyrightText: 2015-2021 Espressif Systems (Shanghai) CO LTD
```
Comment (tore-espressif) [inline components/qma6100p/test/qma6100p_test.c:2]:
```suggestion
 * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
```

File: components/qma6100p/qma6100p.c [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <math.h>
+#include <time.h>
+#include <sys/time.h>
+#include "esp_system.h"
+#include "driver/i2c.h"
```
Comment (tore-espressif) [inline components/qma6100p/qma6100p.c:11]: Leaving this here for discussion:
Do we want to use the new Driver-NG here? with `"driver/i2c_master.h"`?

cc @espzav

File: components/qma6100p/include/qma6100p.h [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+ *
+ * @param sensor object handle of qma6100p
+ * @param isr function to handle interrupts produced by qma6100p
+ *
+ * @return
+ *      - ESP_OK Success
+ *      - ESP_ERR_INVALID_ARG A parameter is NULL or not valid
+ *      - ESP_FAIL Failed to register ISR
+ */
+esp_err_t qma6100p_register_isr(qma6100p_handle_t sensor, const qma6100p_isr_t isr);
```
Comment (tore-espressif) [inline components/qma6100p/include/qma6100p.h:205]: Do we need this function when we pass the isr inside of ` qma6100p_int_config_t->isr` ?

File: components/qma6100p/test/qma6100p_test.c
Comment (tore-espressif) [inline components/qma6100p/test/qma6100p_test.c:1]: How did you run these tests without changes to https://github.com/espressif/esp-bsp/tree/master/test_apps/components ?

File: components/qma6100p/test/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+idf_component_register(SRCS "qma6100p_test.c"
```
Comment (tore-espressif) [inline components/qma6100p/test/CMakeLists.txt:1]: All new components' tests should be a standalone test_apps. eg. like here https://github.com/espressif/esp-bsp/tree/master/components/lcd/esp_lcd_gc9503/test_apps

This way the test application for QMA6110P is self-contained application, which make it more scalable

**Review by tore-espressif — COMMENTED** [review]
@pborcin I quickly went through the code and added some comments.

The code and API overall looks very good! I'll test on HW, once I get the new S3-EYE with QMA6100!

File: components/qma6100p/include/qma6100p.h (hunk 4 above)
Comment (pborcin) [reply components/qma6100p/include/qma6100p.h]: Acts only as a helper function, removed from qma6100p.h file.

**Review by tore-espressif — APPROVED** [review]

**Comment by tore-espressif** [comment]
@pborcin LGTM, but it seems that this step is still missing
image

(I'm sorry, I've also rebased your branch, so please update your local branch too)

**Pull request merged**